apps/emqx_authn/src/simple_authn/emqx_authn_mnesia.erl

```diff
@@ -288,35 +287,25 @@ lookup_user(UserID, #{user_group := UserGroup}) ->
 
 list_users(QueryString, #{user_group := UserGroup}) ->
     NQueryString = QueryString#{<<"user_group">> => UserGroup},
-    emqx_mgmt_api:node_query(node(), NQueryString, ?TAB, ?AUTHN_QSCHEMA, ?QUERY_FUN).
-
-%%--------------------------------------------------------------------
-%% Query Functions
-
-query(Tab, {QString, []}, Continuation, Limit) ->
-    Ms = ms_from_qstring(QString),
-    emqx_mgmt_api:select_table_with_count(
-        Tab,
-        Ms,
-        Continuation,
-        Limit,
-        fun format_user_info/1
-    );
-query(Tab, {QString, FuzzyQString}, Continuation, Limit) ->
-    Ms = ms_from_qstring(QString),
-    FuzzyFilterFun = fuzzy_filter_fun(FuzzyQString),
-    emqx_mgmt_api:select_table_with_count(
-        Tab,
-        {Ms, FuzzyFilterFun},
-        Continuation,
-        Limit,
-        fun format_user_info/1
+    emqx_mgmt_api:node_query(
```

Comment:
Nit: (Not related to the PR directly) These parts of the authenticators' code have always made me confused: here in a "deeply internal and data-related module" we have to deal with both presentation-related (formatting) and controller-related  (qs -> ms) logic... and also call `emqx_mgmt_api` app. But I myself do not have any good ideas how to easily refactor this (and should we do this at all) 😅

Reply:
same for me, it is not easy to understand and use these API. I personally prefer to connect some atomic functions by pipe

Reply:
It is mainly due to implementing the behavior defined in `emqx_authentication.erl`.
:thinking:  Perhaps the user management part can be detached from the behavior of emqx_authentication